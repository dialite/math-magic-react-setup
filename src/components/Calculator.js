/* eslint-disable operator-linebreak */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prefer-stateless-function */
import React, { useState } from 'react';
import '../App.css';
import calculate from '../logic/calculate';
import '../logic/operate';

const Calculator = () => {
  const [data, setData] = useState({
    total: 0,
    next: null,
    operation: null,
    err: null,
  });

  const clickHandler = (e) => {
    setData((prevData) => calculate(prevData, e.target.name));
  };

  return (
    <div className="main-container">
      <h2>Lets Do Some Math</h2>
      <div className="calculation-field">
        <h1 className="answer" data-testid="testH2">
          {data.next || data.operation || data.total || data.err || 0}
        </h1>
        <div className="numbers-and-symbols">
          <button
            type="button"
            onClick={clickHandler}
            className="AC size"
            name="AC"
          >
            AC
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="plus-or-minus size"
            name="+/-"
          >
            +/-
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="percentage size"
            name="%"
          >
            %
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="division size orabg-bg"
            name="รท"
          >
            รท
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="seven size"
            name="7"
          >
            7
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="eight size"
            name="8"
          >
            8
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="nine size"
            name="9"
          >
            9
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="cross size orabg-bg"
            name="x"
          >
            x
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="four size"
            name="4"
          >
            4
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="five size"
            name="5"
          >
            5
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="six size"
            name="6"
          >
            6
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="minus size orabg-bg"
            name="-"
          >
            -
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="one size"
            name="1"
          >
            1
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="two size"
            name="2"
          >
            2
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="three size"
            name="3"
          >
            3
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="plus size orabg-bg"
            name="+"
          >
            +
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="zero"
            name="0"
          >
            0
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="dot size"
            name="."
          >
            .
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="equal size orabg-bg"
            name="="
          >
            =
          </button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
