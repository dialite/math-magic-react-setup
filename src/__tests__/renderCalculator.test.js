import renderer from 'react-test-renderer';
import Operate from '../logic/operate';
import Calculator from '../components/Calculator';

test('Render Calculator', () => {
  const renderCalculator = renderer.create(<Calculator />).toJSON();
  expect(renderCalculator).toMatchSnapshot();
});

test('Multiply 1 x 2 result will be 2', () => {
  const result = Operate(1, 2, 'x');
  expect(result).toBe('2');
});
