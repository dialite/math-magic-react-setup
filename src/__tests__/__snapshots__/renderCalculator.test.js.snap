// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Calculator 1`] = `
<div
  className="main-container"
>
  <h2>
    Lets Do Some Math
  </h2>
  <div
    className="calculation-field"
  >
    <h1
      className="answer"
      data-testid="testH2"
    >
      0
    </h1>
    <div
      className="numbers-and-symbols"
    >
      <button
        className="AC size"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="plus-or-minus size"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="percentage size"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="division size orabg-bg"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="seven size"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="eight size"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="nine size"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="cross size orabg-bg"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="four size"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="five size"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="six size"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="minus size orabg-bg"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="one size"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="two size"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="three size"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="plus size orabg-bg"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="dot size"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="equal size orabg-bg"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
